vignettes/queries_dataset.Rmd

```diff
@@ -0,0 +1,178 @@
+
+
+To help with the derivation of these variables, the admiral function `derive_vars_query()` can be used.
+This function takes as input the dataset (`dataset`) where the grouping must occurs (e.g `ADAE`) and
+a dataset containing the required information to perform the derivation of the grouping variables 
+(`dataset_queries`).
+
+Because the definition, collection and storage of the medical concepts is company specific, the dataset passed to the `dataset_queries` argument of the `derive_vars_query()` function should be created outside of admiral in a specific format so that it can be passed in the function.
+
+This vignette describes the expected structure and content of the dataset passed to the
+`dataset_queries` argument in the `derive_vars_query()` function. 
+
+# Structure of the Query Dataset.
+
+## Variables
+
+Variable | Scope |  Type | Example Value
+------- | ----- | ------ | ----- | ----- | -----
+**VAR_PREFIX** | The prefix used to define the grouping variables |  Character |  "SMQ01" 
+**QUERY_NAME** | The value provided to the grouping variables name|   Character | "Immune-Mediated Guillain-Barre Syndrome"
+**TERM_LEVEL** | The variable used to define the grouping. Used in conjunction with TERM_NAME |  Character | "AEDECOD"
+**TERM_NAME** | A term used to define the grouping. Used in conjunction with TERM_LEVEL |  Character | "GUILLAIN-BARRE SYNDROME"
+**TERM_ID** | A code used to define the grouping. Used in conjunction with TERM_LEVEL |  Integer | 10018767
+QUERY_ID | Id. number of the query. This could be a SMQ identifier |  Integer | 20000131
+QUERY_SCOPE | For SMQs, scope (Broad/Narrow) of the query |  Character | BROAD, NARROW, NA
+QUERY_SCOPE_NUM | For SMQs, scope (Broad/Narrow) of the query |  Integer | 1, 2, NA
+
+**Bold  variables** are required in `dataset_queries`: an error is issued if any of these variables is missing. Other variables are optional.
+
+## Required Content
+
+Each row must be unique within the dataset.
+
+As described above, the variables `VAR_PREFIX`, `QUERY_NAME`, `TERM_LEVEL`, `TERM_NAME` and `TERM_ID` are required.
+The combination of these variables will allow the creation of the grouping variable.
+
+### Input
+
+  + `VAR_PREFIX` must be a character string starting with 2 or 3 letters, followed by a 2-digits number (e.g. "CQ01").
+
+  + `QUERY_NAME` must be a non missing character string and it must be unique within `VAR_PREFIX`. 
+
+  + `TERM_LEVEL` must be a non missing character string. 
+  
+      Each value in `TERM_LEVEL` represents a variable from `dataset` used to define the grouping variables (e.g. `AEDECOD`,`AEBODSYS`, `AELLTCD`).
+    The function `derive_vars_query()` will check that each value given in `TERM_LEVEL` has a corresponding variable in the input  `dataset` and issue an error otherwise.
+
```

Comment:
Do the variable specified under `TERM_LEVEL` need to be the same one within each `VAR_PREFIX`?

Reply:
No, you can have different variables specified in `TERM_LEVEL` within one `VAR_PREFIX`. I have added some details in the input section and extended the example to show this.